crates/ibc/src/events.rs

```diff
@@ -261,6 +261,14 @@ impl TryFrom<IbcEvent> for abci::Event {
     }
 }
 
+impl TryFrom<&IbcEvent> for abci::Event {
+    type Error = Error;
+
+    fn try_from(event: &IbcEvent) -> Result<Self, Self::Error> {
+        abci::Event::try_from(event.clone())
```

Comment:
My original idea was to *replace* `TryFrom<IbcEvent>` with `TryFrom<&IbcEvent>`. Which would in turn mean that every `Event` will also need to change their conversion (e.g. [here](https://github.com/cosmos/ibc-rs/blob/d2435e6f50bfa06c1bd83ccfd0d68b69e7522793/crates/ibc/src/core/ics02_client/events.rs#L200) would become `From<&CreateClient> for abci::Event`).

I opened this after realizing that we never need to take ownership to convert (e.g. [the `ClientAttribute` converts](https://github.com/cosmos/ibc-rs/blob/d2435e6f50bfa06c1bd83ccfd0d68b69e7522793/crates/ibc/src/core/ics02_client/events.rs#L46-L50) the `String` back to `&str`).

This will then be a much bigger change; let's wait to do it after we're done with #279, which will remove about half the code in the repo. We should be done with it some time next week.

Reply:
#279 is merged; would you still like to try this solution out? It's a larger change than you originally thought, so don't feel the need to do it; we could just close this PR if you're not up for it!

Reply:
Done!